### ISSUE TITLE
model

### ISSUE THREAD

**Issue by Yixiaochuan628** [hachreak/movad#2]
Hello! I would like to ask why, when I run your code, all the outputs are positive samples and the AUC is around 0.5. I haven't found where the error is, and I suspect it's a problem with the model.could you give some advice？
thank you！

**Comment by hachreak** [comment]
Hi @Yixiaochuan628 ,
sorry for the long delay for the reply and thank you very much to show interest on our model.
Could you please explain me better the problem? Thanks

**Comment by Yixiaochuan628** [comment]
The issue I encountered is that the outputs of my model have incorrect values, resulting in situations where all outputs are either positive samples or negative samples, while the targets contain both 0s and 1s, which is the expected behavior. The loss function does not decrease during the training process, and I am unable to identify the cause of the problem. Moreover, I haven't made any modifications to your code, and the results I obtained are identical to the images I provided. I would greatly appreciate it if you could help me clarify this confusion.发自我的iPhone------------------ Original ------------------From: Leonardo Rossi  ***@***.***>Date: Mon,May 15,2023 6:57 PMTo: hachreak/movad ***@***.***>Cc: Yixiaochuan628 ***@***.***>, Mention ***@***.***>Subject: Re: [hachreak/movad] model (Issue #2)
Hi @Yixiaochuan628 ,
sorry for the long delay for the reply and thank you very much to show interest on our model.
Could you please explain me better the problem? Thanks

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/hachreak/movad/issues/2#issuecomment-1547636584",
"url": "https://github.com/hachreak/movad/issues/2#issuecomment-1547636584",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by hachreak** [comment]
Could you describe better which model are you traiining and how?
For example:
 - which config file
 - for how many epochs
 - ...

These could be very useful info to understand what is went wrong, thanks.

**Comment by Yixiaochuan628** [comment]
The model is v4-1，i run about 200 epoch.Is it because there is no convergence? I also ran the v1-1 model, and the output is only 0 or sometimes only 1. I ran on the 3080 for almost two days, but the loss did not decrease. When running the test, the auc is 0.5.发自我的iPhone------------------ Original ------------------From: Leonardo Rossi  ***@***.***>Date: Wed,May 17,2023 9:54 PMTo: hachreak/movad ***@***.***>Cc: Yixiaochuan628 ***@***.***>, Mention ***@***.***>Subject: Re: [hachreak/movad] model (Issue #2)
Could you describe better which model are you traiining and how?
For example:

which config file
for how many epochs
...

These could be very useful info to understand what is went wrong, thanks.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/hachreak/movad/issues/2#issuecomment-1551443226",
"url": "https://github.com/hachreak/movad/issues/2#issuecomment-1551443226",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by hachreak** [comment]
Because we made a code cleaning and refactoring before publish the code, I'm trying to retrain the model to try reproduce your problem and check everything is working properly.
Please, be patient and I'll tell you more. Thanks :)

**Comment by Yixiaochuan628** [comment]
Thank you for your sincere help in resolving my doubts.i express my high respect to you.发自我的iPhone------------------ Original ------------------From: Leonardo Rossi  ***@***.***>Date: Thu,May 18,2023 4:30 PMTo: hachreak/movad ***@***.***>Cc: Yixiaochuan628 ***@***.***>, Mention ***@***.***>Subject: Re: [hachreak/movad] model (Issue #2)
Because we made a code cleaning and refactoring before publish the code, I'm trying to retrain the model to try reproduce your problem and check everything is working properly.
Please, be patient and I'll tell you more. Thanks :)

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/hachreak/movad/issues/2#issuecomment-1552720537",
"url": "https://github.com/hachreak/movad/issues/2#issuecomment-1552720537",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by hachreak** [comment]
I found a [bug](https://github.com/hachreak/movad/commit/72ef9aeb428e2f1fabd24f30abe5b5410e749c05) on collecting the outputs, introduced on code refactoring. 
So, I'm still waiting the evaluation to check if now it works again as expected.

**Comment by Yixiaochuan628** [comment]
That's great, and I would appreciate it if you could send me the modified code发自我的iPhone------------------ Original ------------------From: Leonardo Rossi  ***@***.***>Date: Fri,May 19,2023 4:02 PMTo: hachreak/movad ***@***.***>Cc: Yixiaochuan628 ***@***.***>, Mention ***@***.***>Subject: Re: [hachreak/movad] model (Issue #2)
I found a bug on collecting the outputs, introduced on code refactoring.
So, I'm still waiting the evaluation to check if now it works again as expected.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/hachreak/movad/issues/2#issuecomment-1554197482",
"url": "https://github.com/hachreak/movad/issues/2#issuecomment-1554197482",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by hachreak** [comment]
I already updated the repository.
You can simply merge fast forward from github repo.

**Comment by Yixiaochuan628** [comment]
thanks again发自我的iPhone------------------ Original ------------------From: Leonardo Rossi  ***@***.***>Date: Fri,May 19,2023 5:55 PMTo: hachreak/movad ***@***.***>Cc: Yixiaochuan628 ***@***.***>, Mention ***@***.***>Subject: Re: [hachreak/movad] model (Issue #2)
I already updated the repository.
You can simply merge fast forward from github repo.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/hachreak/movad/issues/2#issuecomment-1554324582",
"url": "https://github.com/hachreak/movad/issues/2#issuecomment-1554324582",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by Yixiaochuan628** [comment]
I apologize for bothering you again. After adding dtype=float, my loss is still maintaining at a high level, and the neg and pos also seem not quite normal. I'm still training the network, which could possibly be because it hasn't converged yet.can you tell me where the problem lies？发自我的iPhone------------------ Original ------------------From: Leonardo Rossi  ***@***.***>Date: Fri,May 19,2023 5:55 PMTo: hachreak/movad ***@***.***>Cc: Yixiaochuan628 ***@***.***>, Mention ***@***.***>Subject: Re: [hachreak/movad] model (Issue #2)
I already updated the repository.
You can simply merge fast forward from github repo.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/hachreak/movad/issues/2#issuecomment-1554324582",
"url": "https://github.com/hachreak/movad/issues/2#issuecomment-1554324582",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by hachreak** [comment]
Actually, on training doesn't change too much (only computing the percentage of correct answer).
But on evaluation change a lot, because it saves the results as float value instead of long, preserving the correct value and not translating it to 0/1.

You can see my ongoing training:

```
Epoch: 450 / 200: 100%|██████████| 1376/1376 [1:24:13<00:00,  3.67s/it]
save file ./output/v4_1/eval/results-450.pkl
load file ./output/v4_1/eval/results-450.pkl

[Correctness] f-AUC = 0.79058
             PR-AUC = 0.68929
           F1-Score = 0.63327
           F1-Mean  = 0.70144
           Accuracy = 0.72977
```

I can't show the loss value because the default config debug_loss was `False`, but it should go down during the training.

**Comment by Yixiaochuan628** [comment]
Thank you for your answer, I have learned a lot from it.发自我的iPhone------------------ Original ------------------From: Leonardo Rossi  ***@***.***>Date: Mon,May 22,2023 4:29 PMTo: hachreak/movad ***@***.***>Cc: Yixiaochuan628 ***@***.***>, Mention ***@***.***>Subject: Re: [hachreak/movad] model (Issue #2)
Actually, on training doesn't change too much (only computing the percentage of correct answer).
But on evaluation change a lot, because it saves the results as float value instead of long, preserving the correct value and not translating it to 0/1.
You can see my ongoing training:
Epoch: 450 / 200: 100%|██████████| 1376/1376 [1:24:13<00:00,  3.67s/it]
save file ./output/v4_1/eval/results-450.pkl
load file ./output/v4_1/eval/results-450.pkl

[Correctness] f-AUC = 0.79058
             PR-AUC = 0.68929
           F1-Score = 0.63327
           F1-Mean  = 0.70144
           Accuracy = 0.72977

I can't show the loss value because the default config debug_loss was False, but it should go down during the training.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/hachreak/movad/issues/2#issuecomment-1556774554",
"url": "https://github.com/hachreak/movad/issues/2#issuecomment-1556774554",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by hachreak** [comment]
The bug was fixed (and [pretrained](https://github.com/IMPLabUniPr/movad/releases/tag/v1.0) are now available).
